src/common/services/auth.js

```diff
@@ -0,0 +1,108 @@
+/*
+** This service is used to authenticate users 
+** author: @ishmeet-kuliza
+*/
+angular.module('Vyomo')
+.factory('auth', ['$http', '$q', 'env', function($http, $q, env) {
+
+  var userObj,
+      BASE_URL = env.BASE_URL;
+  /**
+  * Set common request headers
+  */
+  function setDefaultHeaders () {
+    // set default headers for all outgoing requests
+    // needed this for patch.. as in angular patch does not have content-type set by default
+    $http.defaults.headers.common['Content-Type'] = 'text/plain;charset=utf-8';
+    $http.defaults.headers.common['Accept'] = 'text/plain';
+    window.console.log('THis is env', env.hello);
+  }
+
+  setDefaultHeaders();
+
+  function _generateUserObj(data) {
+    return {
+      sessionToken: data.sessionToken,
+      'otpVerified': data.otp_verified,
+      'otp': data.otp //For testing purposes
+    };
+  }
+
+  function authenticate(number, password) {
+    var deferred = $q.defer();
+    // login call to server.
+    var authUrl = BASE_URL + '/signin';
+
+    $http.post(authUrl, { number: number, password: password }).then(function(response) {
+      if (response && response.data && response.data.status_code === 200){
+        setDefaultHeaders();
+        userObj = _generateUserObj(response.data.message);
+        // persist in session storage so this is available for page refreshes.
+        window.sessionStorage.userObj = JSON.stringify(userObj);
```

Comment:
how userObj is persisted ?. by this user will be logged out when browser is closed.
plzz explain